### Performing Arithmetic Operations Using Python

In [1]:
2+3+9

14

In [2]:
99-73

26

In [3]:
23.54 * -1432

-33709.28

In [4]:
100/7 # Complete quotient with decimal too

14.285714285714286

In [5]:
100// 7 # Only the quotient as an Integer

14

In [6]:
100 % 7 # Displays the remainder

2

In [7]:
5 **3 # raise to the power

125

As you might expect, operators like `/` and `*` take precedence over other operators like `+` and `-` as per mathematical conventions. You can use parentheses, i.e. `(` and `)`, to specify the order in which operations are performed.

In [8]:
((2 + 5) * (17 - 3)) / (4**3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 

In [3]:
# 1.The population of a town is 198568. Out of them 45312 are men and 35678 are women. Find the number of children
# in the town.
(198568 - (45312 + 35678))

117578

In [4]:
# 2. A shopkeeper has 2425 boxes of 24 pencils each. How many pencils do all the boxes have in all?
(24 * 2425)

58200

In [5]:
# 3. Linda bought a coat for $2265 and a saree for $2150. She gave $5000 to the shopkeeper. 
# How much money did the shopkeeper return to her?
(5000 - (2265 + 2150))

585

In [8]:
# 13. Maria bought 96 toys priced equally for $12960. The amount of $1015 is still left with her. Find the cost of 
# each toy and the amount she had.
print(12960/96)
print(12960 + 1015)

135.0
13975


## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [9]:
(500 * (0.2*1.25))

125.0

The variables `cost_of_ice_bag`, `profit_margin`, and `number_of_bags` now contain the information provided in the word problem. We can check the value of a variable by typing its name into a cell. We can combine variables using arithmetic operations to create other variables.

In [10]:
cost_of_ice_bag = 1.25

In [11]:
profit_margin = 0.2

In [12]:
number_of_bags = 500

In [13]:
profit_margin

0.2

In [14]:
profit_per_bag = (profit_margin * cost_of_ice_bag)

In [15]:
profit_per_bag

0.25

In [16]:
total_profit = profit_per_bag * number_of_bags

In [17]:
total_profit

125.0

In [18]:
print("The grocery store makes a total profit of ${}".format(total_profit))

The grocery store makes a total profit of $125.0


Let's rewrite the solution to our word problem within a single cell.

In [1]:
# input vars of store
cost_of_ice_bag = 1.25
profit_margin = 0.2
no_of_bags = 500

# calulations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = profit_per_bag * no_of_bags

# result
print("The grocery store makes a total profit of ${}".format(total_profit))

The grocery store makes a total profit of $125.0


Inline and single line comments start with `#`, whereas multi-line comments begin and end with three quotes i.e `"""`. Here are some examples of code comments:

In [2]:
my_favourite_number = 1 # and inline comment

In [3]:
# This comment gets its own line
my_least_favorite_number = 3

In [4]:
""" This is a
multi-line comment"""
a_neutral_number = 5

> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [5]:
# Question inputs
total_operating_cost = 5000
no_of_people = 29
price_per_ticket = 200

# operations
total_money_earned = no_of_people * price_per_ticket
net_profit = total_money_earned - total_operating_cost

# Output
print("Total profit made by the company: ${}".format(net_profit))

Total profit made by the company: $800


## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [6]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [7]:
# Equality Check - True
my_favorite_number == 1

True

In [8]:
# Equality Check - False
my_favorite_number == my_least_favorite_number

False

In [9]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [10]:
# Not equal check - False
a_neutral_number != 3

False

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [14]:
cost_icebag = 1.25
is_icebag_expensive = cost_icebag >=10
print("Is the ice bag expensive?",is_icebag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

In [15]:
my_favorite_number

1

In [16]:
my_favorite_number > 0 and my_favorite_number <=3

True

In [17]:
my_favorite_number < 0 and my_favorite_number >=3

False

In [18]:
True and False

False

In [19]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [20]:
a_neutral_number

3

In [21]:
a_neutral_number == 3 or my_favorite_number <0

True

In [22]:
a_neutral_number != 3 or my_favorite_number <0

False

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [23]:
not a_neutral_number == 3 

False

In [24]:
not False

True

In [25]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [26]:
(2 > 3 and 4 <= 5) or not (my_favorite_number <0 and True)

True

In [27]:
not (True and 0<1) or (False and True)

False